Added lesson_07

MarianaKhl opened the pull request.

MarianaKhl committed:
Added lesson_07

edanilovets commented on lesson_07/homework_07.py:
```diff
@@ -0,0 +1,185 @@
+    # Complete the while loop condition.
+    while True:
+        result = number * multiplier
+        if  result <= 25:
```
```suggestion
        if result <= 25:
```

edanilovets commented on lesson_07/homework_07.py:
```diff
@@ -0,0 +1,185 @@
+
+multiplication_table(3)
+
+print("#"*100)
```
```suggestion
print("#" * 100)
```

edanilovets commented on lesson_07/homework_07.py:
```diff
@@ -0,0 +1,185 @@
+def result_sum(number1, number2):
+    sum = number1 + number2
+    print(sum)
+result_sum(10,12)
```
```suggestion
result_sum(10, 12)
```

edanilovets commented on lesson_07/homework_07.py:
```diff
@@ -0,0 +1,185 @@
+    if len(sentance) == 0:
+        return f"The sentence are absent"
+    words = sentance.split()   # is the list of words obtained after splitting the sentence with split()
+    longest_word = max(words, key= len)   # calculates the length of each word
```
```suggestion
    longest_word = max(words, key=len)   # calculates the length of each word
```

edanilovets commented on lesson_07/homework_07.py:
```diff
@@ -0,0 +1,185 @@
+sentance = "Hi! How are you?"
+vice_versa_result(sentance)
+
+print("#"*100)
```
```suggestion
print("#" * 100)
```

edanilovets commented on lesson_07/homework_07.py:
```diff
@@ -0,0 +1,185 @@
+    average_result = sum(numbers) / len(numbers)
+    print(average_result)
+
+numbers = [1,6,33]
```
```suggestion
numbers = [1, 6, 33]
```

edanilovets commented on lesson_07/homework_07.py:
```diff
@@ -0,0 +1,185 @@
+
+def vice_versa_result(sentence):
+    if len(sentence) == 0:
+        return f"The sentence are absent"
```
```suggestion
        return f"The sentence is empty"
```

edanilovets commented on lesson_07/homework_07.py:
```diff
@@ -0,0 +1,185 @@
+        return f"The sentence are absent"
+
+    reverce_line = sentence[::-1]
+    print(f"Vice versa result: {reverce_line}")
```
```suggestion
    print(f"Reversed result: {reverce_line}")
```

edanilovets commented on lesson_07/homework_07.py:
```diff
@@ -0,0 +1,185 @@
+def search_h_letter():
+
+    while True:
+        input_string = input("Enter the any word with letter 'h' , please: ").lower()
```
```suggestion
        input_string = input("Enter any word with letter 'h' , please: ").lower()
```

MarianaKhl committed:
Added changes to the lesson_07

MarianaKhl commented:
Hi!

Added changes to the lesson_07/homework-07:
https://github.com/MarianaKhl/Lesson_2-GitHub-/pull/10

On Thu, Sep 5, 2024 at 1:53 PM Eugene Danilovets ***@***.***>
wrote:

> ***@***.**** commented on this pull request.
> ------------------------------
>
> In lesson_07/homework_07.py
> <https://github.com/MarianaKhl/Lesson_2-GitHub-/pull/10#discussion_r1745318508>
> :
>
> > @@ -0,0 +1,185 @@
> +# task 1
> +""" Задача - надрукувати табличку множення на задане число, але
> +лише до максимального значення для добутку - 25.
> +Код майже готовий, треба знайти помилки та випраавити\доповнити.
> +"""
> +def multiplication_table(number):
> +    # Initialize the appropriate variable
> +    multiplier = 1
> +
> +    # Complete the while loop condition.
> +    while True:
> +        result = number * multiplier
> +        if  result <= 25:
>
> ⬇️ Suggested change
>
> -        if  result <= 25:
> +        if result <= 25:
>
> ------------------------------
>
> In lesson_07/homework_07.py
> <https://github.com/MarianaKhl/Lesson_2-GitHub-/pull/10#discussion_r1745319067>
> :
>
> > +def multiplication_table(number):
> +    # Initialize the appropriate variable
> +    multiplier = 1
> +
> +    # Complete the while loop condition.
> +    while True:
> +        result = number * multiplier
> +        if  result <= 25:
> +            print(f"{number} x {multiplier} = {result}")
> +            multiplier += 1
> +        else:
> +            break
> +
> +multiplication_table(3)
> +
> +print("#"*100)
>
> ⬇️ Suggested change
>
> -print("#"*100)
> +print("#" * 100)
>
> ------------------------------
>
> In lesson_07/homework_07.py
> <https://github.com/MarianaKhl/Lesson_2-GitHub-/pull/10#discussion_r1745319202>
> :
>
> > +            print(f"{number} x {multiplier} = {result}")
> +            multiplier += 1
> +        else:
> +            break
> +
> +multiplication_table(3)
> +
> +print("#"*100)
> +
> +# task 2
> +"""  Написати функцію, яка обчислює суму двох чисел.
> +"""
> +def result_sum(number1, number2):
> +    sum = number1 + number2
> +    print(sum)
> +result_sum(10,12)
>
> ⬇️ Suggested change
>
> -result_sum(10,12)
> +result_sum(10, 12)
>
> ------------------------------
>
> In lesson_07/homework_07.py
> <https://github.com/MarianaKhl/Lesson_2-GitHub-/pull/10#discussion_r1745320029>
> :
>
> > +    print(f"Vice versa result: {reverce_line}")
> +
> +sentance = "Hi! How are you?"
> +vice_versa_result(sentance)
> +
> +print("#"*100)
> +
> +# task 5
> +"""  Написати функцію, яка приймає список слів та повертає найдовше слово у списку.
> +"""
> +
> +def long_word(sentance):
> +    if len(sentance) == 0:
> +        return f"The sentence are absent"
> +    words = sentance.split()   # is the list of words obtained after splitting the sentence with split()
> +    longest_word = max(words, key= len)   # calculates the length of each word
>
> ⬇️ Suggested change
>
> -    longest_word = max(words, key= len)   # calculates the length of each word
> +    longest_word = max(words, key=len)   # calculates the length of each word
>
> ------------------------------
>
> In lesson_07/homework_07.py
> <https://github.com/MarianaKhl/Lesson_2-GitHub-/pull/10#discussion_r1745320199>
> :
>
> > +
> +# task 4
> +"""  Написати функцію, яка приймає рядок та повертає його у зворотному порядку.
> +"""
> +
> +def vice_versa_result(sentence):
> +    if len(sentence) == 0:
> +        return f"The sentence are absent"
> +
> +    reverce_line = sentence[::-1]
> +    print(f"Vice versa result: {reverce_line}")
> +
> +sentance = "Hi! How are you?"
> +vice_versa_result(sentance)
> +
> +print("#"*100)
>
> ⬇️ Suggested change
>
> -print("#"*100)
> +print("#" * 100)
>
> ------------------------------
>
> In lesson_07/homework_07.py
> <https://github.com/MarianaKhl/Lesson_2-GitHub-/pull/10#discussion_r1745320507>
> :
>
> > +result_sum(10,12)
> +
> +print("#"*100)
> +
> +# task 3
> +"""  Написати функцію, яка розрахує середнє арифметичне списку чисел.
> +"""
> +
> +def result_average(numbers):
> +    if len(numbers) == 0:
> +        return f"The list is empty"
> +
> +    average_result = sum(numbers) / len(numbers)
> +    print(average_result)
> +
> +numbers = [1,6,33]
>
> ⬇️ Suggested change
>
> -numbers = [1,6,33]
> +numbers = [1, 6, 33]
>
> ------------------------------
>
> In lesson_07/homework_07.py
> <https://github.com/MarianaKhl/Lesson_2-GitHub-/pull/10#discussion_r1745321629>
> :
>
> > +
> +    average_result = sum(numbers) / len(numbers)
> +    print(average_result)
> +
> +numbers = [1,6,33]
> +result_average(numbers)
> +
> +print("#"*100)
> +
> +# task 4
> +"""  Написати функцію, яка приймає рядок та повертає його у зворотному порядку.
> +"""
> +
> +def vice_versa_result(sentence):
> +    if len(sentence) == 0:
> +        return f"The sentence are absent"
>
> ⬇️ Suggested change
>
> -        return f"The sentence are absent"
> +        return f"The sentence is empty"
>
> ------------------------------
>
> In lesson_07/homework_07.py
> <https://github.com/MarianaKhl/Lesson_2-GitHub-/pull/10#discussion_r1745322346>
> :
>
> > +
> +numbers = [1,6,33]
> +result_average(numbers)
> +
> +print("#"*100)
> +
> +# task 4
> +"""  Написати функцію, яка приймає рядок та повертає його у зворотному порядку.
> +"""
> +
> +def vice_versa_result(sentence):
> +    if len(sentence) == 0:
> +        return f"The sentence are absent"
> +
> +    reverce_line = sentence[::-1]
> +    print(f"Vice versa result: {reverce_line}")
>
> ⬇️ Suggested change
>
> -    print(f"Vice versa result: {reverce_line}")
> +    print(f"Reversed result: {reverce_line}")
>
> ------------------------------
>
> In lesson_07/homework_07.py
> <https://github.com/MarianaKhl/Lesson_2-GitHub-/pull/10#discussion_r1745323187>
> :
>
> > +lst1 = ['1', '2', 3, True, 'False', 5, '6', 7, 8, 'Python', 9, 0, 'Lorem Ipsum']
> +creator_string_list(lst1)
> +
> +print("#"*100)
> +
> +
> +# task 9
> +
> +""""  # Напишіть цикл, який буде вимагати від користувача ввести слово, в якому є літера "h"
> +# (враховуються як великі так і маленькі). Цикл не повинен завершитися,
> +# якщо користувач ввів слово без букви "h".
> +"""
> +def search_h_letter():
> +
> +    while True:
> +        input_string = input("Enter the any word with letter 'h' , please: ").lower()
>
> ⬇️ Suggested change
>
> -        input_string = input("Enter the any word with letter 'h' , please: ").lower()
> +        input_string = input("Enter any word with letter 'h' , please: ").lower()
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/MarianaKhl/Lesson_2-GitHub-/pull/10#pullrequestreview-2282684270>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/BHA2N25OCPI3CVEKZ7FZ5WLZVBA4JAVCNFSM6AAAAABNVCBJZOVHI2DSMVQWIX3LMV43YUDVNRWFEZLROVSXG5CSMV3GSZLXHMZDEOBSGY4DIMRXGA>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>
